django/db/backends/sqlite3/features.py

```diff
@@ -147,6 +147,12 @@ def introspected_field_types(self):
             "SmallAutoField": "AutoField",
         }
 
+    @cached_property
+    def max_query_params(self):
+        # This limit can be lowered per connection at runtime with setlimit().
+        # If set, the cached property must be deleted to take effect.
```

Comment:
The limit function takes nanoseconds to run:

```
In [1]: import sqlite3

In [2]: c = sqlite3.connect(":memory:")

In [3]: %timeit c.getlimit(sqlite3.SQLITE_LIMIT_VARIABLE_NUMBER)
40.1 ns ± 0.0617 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

I think it just returns an in-memory number from some struct.

Therefore, no need to cache:

```suggestion
    @property
    def max_query_params(self):
```

Reply:
Thanks Adam! I'm seeing the same results with an on-disk database as well. I've added a commit to change this and drop the `del connection.features.max_query_params` lines. 

I kept the first line of the comment to help document why we're not using `cached_property`, but if it's clear enough I'm also happy to remove it.

Reply:
Yeah, a comment seems like a good idea.